### PR TITLE
Radial offset feature for #4217

### LINKED ISSUE #4217: Feature request: offset radial origin for polar plots
I'd be really interested in having a nice way to produce polar plots where the origin of the radial axis is offset from the centre, something like this:

origin example

Having the radial offset is particularly useful for plotting polar histograms, which tend to suffer from the problem that long bars are over-emphasised relative to short bars because their areas being much larger due to the distorting effect of the polar transformation. Adding a radial offset helps to reduce the visual impact of this distortion.

There are of course hacky ways to achieve this sort of effect, e.g. offsetting the r coordinates of the data, then manually setting the radial ticks. I also came across [this SO question](http://stackoverflow.com/q/16605137/1461210). The accepted answer, using `floating_axes.GridHelperCurveLinear`, is close to what I'm looking for:

so_example

However, it's still a bit of a nasty hack, since it requires the radial ticks to be set manually.

How much work would it be to add a `set_r_offset()` method to allow one to control the relative radial position of the r-axis origin? I imagine it accepting a float between 0 and 1, where 0 would place the r origin exactly at the centre, and 1 would place it exactly at the edge of the plot.

### PR DESCRIPTION
**Pull request by SynergicR**
Resolves #4217 by adding set_r_offset(), get_r_offset() methods. A set of test cases will be made if solution is agreed upon.

### CODE REVIEW

File: lib/matplotlib/projections/polar.py [hunk 1]
```diff
@@ -686,6 +690,46 @@ def drag_pan(self, button, key, x, y):
...
+        self.set_rmin(offset)
+        self._r_offset = input_offset
+        self.add_artist(Wedge((0.5,0.5), input_offset*.5, 0,360, transform=self.transAxes, color=self.patch.get_facecolor(),zorder=99,ec='k'))
+
+    def get_r_offset(self):
+        """
+        Returns the radial offset
+        
+        Parameters
+        ----------
```
Comment (tacaswell) [inline lib/matplotlib/projections/polar.py]: Don't need the empty docstring section.

File: lib/matplotlib/projections/polar.py [hunk 2]
```diff
@@ -686,6 +690,46 @@ def drag_pan(self, button, key, x, y):
...
+        """
+        Creates an offset from the origin of a polar plot
+        
+        Parameters
+        ----------
+        input_offset : float
+              A number between 0 and 1 where 0 is the center of the plot and 1
+              is the edge of the plot
+        
+        Returns
```
Comment (tacaswell) [inline lib/matplotlib/projections/polar.py]: Don't need the empty returns section.

File: lib/matplotlib/projections/polar.py [hunk 3]
```diff
@@ -686,6 +690,46 @@ def drag_pan(self, button, key, x, y):
...
+            raise NotImplementedError("Offset must be greater than 0 and less than 1.")
+        elif input_offset == 0:
+            return
+        rmax = self.get_rmax()
+        rmin = self.get_rmin()
+        rrange = rmax-rmin
+        offset = rmin-(input_offset*rrange)/(1.0-input_offset)
+        self.set_rmin(offset)
+        self._r_offset = input_offset
+        self.add_artist(Wedge((0.5,0.5), input_offset*.5, 0,360, transform=self.transAxes, color=self.patch.get_facecolor(),zorder=99,ec='k'))
```
Comment (tacaswell) [inline lib/matplotlib/projections/polar.py]: Does this just mask out the center of the graph?  Calling this function more than once will result in an accumulation of artists.

**Comment by tacaswell** [comment]
This should probably be implemented by further warping the transforms, not by masking out the center of the plot.  `zorder` is a float, so no matter what value you put, it can be broken.

**Comment by SynergicR** [comment]
Yes the center is masked out, mainly to cover the excess data after applying the offset as well as grid lines. Would destroying the artist and redrawing, if called multiple times, be viable? If not, then you would prefer the graph to be a 'doughnut' by warping the transforms?

**Comment by tacaswell** [comment]
I don't think doing this by masking is an acceptable solution and if it is implemented it has to be done via the transforms.

The example linked to in #4217 should give you a reasonable start on how to implement this.

**Comment by SynergicR** [comment]
Thank you for the feedback. I'll get to work.

**Comment by tacaswell** [comment]
It is a clever solution, but not something we can include in the main-line.

**Comment by WeatherGod** [comment]
We probably should revisit the negative radius argument again...

**Comment by alimehdi92** [comment]
I am helping @SynergicR with this feature.
We had a question, are we allowed to change the object type? Since in the [SO](http://stackoverflow.com/questions/16605137/matplotlib-polar-plot-radial-axis-offset) question, a "floating_axes" object is created and returned.
Would this be a problem since we are not modifying the actual PolarAxes Object?

**Comment by efiring** [comment]
Floating axes are a somewhat experimental feature in the mpl_toolkits/axisartist toolkit, so we can't use them in mpl itself.  I think that what is needed here is either the addition of functionality to PolarAxes, or a subclass or variant of it that has the desired additional characteristics.  This might not be very complicated in the end, but I expect it will require some detailed study of how PolarAxes works, so that you can figure out what modifications are needed.  I expect the transform will be different, and the patch boundary will have an inside circle as well as an outside circle.  Getting everything right could be quite challenging.

**Comment by tacaswell** [comment]
Closing this because it will not be merged using this method (masking the center).

**Pull request closed without merging**